tests/integration/modules/test_firewall.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+# Import Python libs
+from __future__ import absolute_import
+
+# Import Salt Testing libs
+from tests.support.case import ModuleCase
+from tests.support.unit import skipIf
+from tests.support.helpers import destructiveTest
+
+# Import Salt Libs
+import salt.utils
+
+
+@skipIf(not salt.utils.is_windows(), 'Tests for only Windows')
+class FirewallTest(ModuleCase):
+    '''
+    Validate windows firewall module
+    '''
+    def _pre_firewall_status(self, pre_run):
```

Comment:
So, given this I almost wonder if we should wrap some of the calls below in try/except/finally statements where the finally calls this. That way, if an exception is raised by salt, we'll at least reset the FW for the next test before we go on.